pandas/core/dtypes/dtypes.py

```diff
@@ -893,11 +893,15 @@ def __new__(cls, subtype=None):
                 m = cls._match.search(subtype)
                 if m is not None:
                     subtype = m.group('subtype')
-
-            try:
-                subtype = pandas_dtype(subtype)
-            except TypeError:
-                raise TypeError("could not construct IntervalDtype")
+                else:
+                    # if no match found, a bad datatype was passed
+                    msg = ('category, object, and string subtypes are not '
+                           'supported for IntervalDtype')
+                    raise TypeError(msg)
+        try:
+            subtype = pandas_dtype(subtype)
```

Comment:
Instead of using `pandas_dtype` it might be a bit cleaner to instead explicitly check for allowed subtypes (numeric/datetime/timedelta).  I haven't looked into this in too much detail, so it could be a dead end, but should be a viable path to explore.

Reply:
thanks for the suggestion! i agree this might be a better route.

the latest changes check explicitly for these three subtypes. there may be a cleaner way to accomplish this, though, let me know what you think